R/analyse_strat.R

```diff
@@ -24,34 +29,34 @@
 analyse_strat <- function(data = NULL, Ts, Tb, dates, H_ice = NULL, drho = 0.1, NH = TRUE){
 
   if(!is.null(data)){
-    data[,2] <- abs(data[,2])
-    depths <- unique(data[,2])
+    data[, 2] <- abs(data[, 2])
+    depths <- unique(data[, 2])
     depths <- depths[order(depths)]
 
     # Find closest depth near the surface without NA
-    for(i in 1:length(depths)){
-      Ts = data[data[,2] == depths[i],3]
-      if(sum(is.na(Ts))/length(Ts) < 0.25){
+    for(i in seq_len(length(depths))){
+      Ts <- data[data[, 2] == depths[i], 3]
+      if(sum(is.na(Ts)) / length(Ts) < 0.25){
         if(i != 1){
-          message('Warning: Using ', depths[i], ' as the surface.')
+          message("Warning: Using ", depths[i], " as the surface.")
         }
         break
       }
     }
     # Find closest depth near the bottom without NA
-    for(i in length(depths):1){
-      Tb = data[data[,2] == depths[i],3]
-      if(sum(is.na(Tb))/length(Tb) < 0.25){
+    for(i in seq_len((length(depths)):1)){
```

Comment:
This will go wrong. Change to rev(seq_len(length(depths))), or leave as is.

Reply:
changed it in an additional commit f5c4766eab1f0bf2ccd7cef2bb04277f52ad8789